Source/Models/BrickDimension.swift

```diff
@@ -18,39 +18,106 @@ public struct BrickSize {
     }
 }
 
+public typealias RangeDimensionPair = (dimension: BrickDimension, minimumLength: CGFloat)
+
+public struct BrickRangeDimension {
+    
+    internal var dimensionPairs : [RangeDimensionPair]
+    
+    public init(minimum dimension: BrickDimension, additionalRangePairs: [RangeDimensionPair]?) {
+        dimensionPairs = [(dimension, 0)]
+        if let additionalRangePairs = additionalRangePairs {
+            dimensionPairs.append(contentsOf: additionalRangePairs.sorted {
+                $0.minimumLength < $1.minimumLength
+            })
+        }
+    }
+    
+    public func dimension(forWidth width: CGFloat?) -> BrickDimension {
+        guard let width = width  else {
+            return dimensionPairs.first!.dimension
+        }
+        var dimension = dimensionPairs.first?.dimension
+        for RangeDimensionPair in dimensionPairs {
+            if RangeDimensionPair.minimumLength > width {
+                break
+            }
+            else {
+                dimension = RangeDimensionPair.dimension
+            }
+        }
+        return dimension!
+    }
+}
+
+internal func almostEqualRelative(first: CGFloat, second: CGFloat, maxRelDiff: CGFloat = CGFloat.ulpOfOne) -> Bool
+{
+    // Calculate the difference.
+    let diff = fabs(first - second);
+    let a = fabs(first)
+    let b = fabs(second)
+    // Find the largest
+    let largest = (b > a) ? b : a
+    
+    if diff <= largest * maxRelDiff {
+        return true
+    }
+    return false
+}
+
+public func ==(lhs: RangeDimensionPair, rhs: RangeDimensionPair) -> Bool {
+    return lhs.dimension == rhs.dimension && almostEqualRelative(first: lhs.minimumLength, second: rhs.minimumLength)
+}
+
+public func ==(lhs: BrickRangeDimension, rhs: BrickRangeDimension) -> Bool {
+    if lhs.dimensionPairs.count != rhs.dimensionPairs.count {
+        return false
+    }
+    for (index, value) in lhs.dimensionPairs.enumerated() {
+        if value != rhs.dimensionPairs[index] {
+            return false
+        }
+    }
+    return true
+}
+
+
 public enum BrickDimension {
     case ratio(ratio: CGFloat)
     case fixed(size: CGFloat)
     case fill
-
+    
     indirect case auto(estimate: BrickDimension)
     indirect case orientation(landscape: BrickDimension, portrait: BrickDimension)
     indirect case horizontalSizeClass(regular: BrickDimension, compact: BrickDimension)
     indirect case verticalSizeClass(regular: BrickDimension, compact: BrickDimension)
-
+    indirect case dimensionRange(minimum: BrickDimension, additionalRangePairs: [RangeDimensionPair]?)
+    
     public var isEstimate: Bool {
-        switch self.dimension {
+        switch self.dimension(withValue: nil) {
         case .auto(_): return true
         default: return false
         }
     }
 
-    var dimension: BrickDimension {
+    func dimension(withValue value: CGFloat? = nil) -> BrickDimension {
```

Comment:
The concern here is that only one case actually needs to use a value. We'd be forcing people to provide a completely unused value for all the other cases.